src/libraries/System.Text.Json/tests/Serialization/TestClasses.NonGenericCollections.cs

```diff
@@ -180,6 +185,11 @@ public IEnumerator GetEnumerator()
         }
     }
 
+    public class WrapperForICollectionPrivateConstructor : WrapperForICollection
+    {
+        private WrapperForICollectionPrivateConstructor() { }
```

Comment:
How about adding internal ctors as well?

Reply:
added classes and tests for internal ctors in fe06beb